src/Google.Api.Gax/ApiCallRetryExtensions.cs

```diff
@@ -0,0 +1,103 @@
+﻿using Google.Protobuf;
+using Grpc.Core;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Google.Api.Gax
+{
+    public static class ApiCallRetryExtensions
```

Comment:
We should probably discuss the layout/naming of the extension methods at some point, but that can come later. (It'll be a binary breaking change, but not a source one.)